File: sdkwork-backend-sdk-csharp/Api/InvitationCodeApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Backend.Http;
using Backend.Models;

namespace Backend.Api
{
    public class InvitationCodeApi
    {
        private readonly HttpClient _client;

        public InvitationCodeApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Update an existing invitation code
        /// </summary>
        public async Task<PlusApiResultInvitationCodeVO?> UpdateAsync(InvitationCodeForm body)
        {
            return await _client.PutAsync<PlusApiResultInvitationCodeVO>(ApiPaths.BackendPath("/invitation/code"), body);
        }

        /// <summary>
        /// Create a new invitation code
        /// </summary>
        public async Task<PlusApiResultInvitationCodeVO?> CreateAsync(InvitationCodeForm body)
        {
            return await _client.PostAsync<PlusApiResultInvitationCodeVO>(ApiPaths.BackendPath("/invitation/code"), body);
        }

        /// <summary>
        /// Get invitation codes by page
        /// </summary>
        public async Task<PlusApiResultPageInvitationCodeVO?> ListByPageAsync(QueryListForm? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<PlusApiResultPageInvitationCodeVO>(ApiPaths.BackendPath("/invitation/code/list"), body, query);
        }

        /// <summary>
        /// Get all invitation codes
        /// </summary>
        public async Task<PlusApiResultListInvitationCodeVO?> ListAllEntitiesAsync(QueryListForm? body = null)
        {
            return await _client.PostAsync<PlusApiResultListInvitationCodeVO>(ApiPaths.BackendPath("/invitation/code/list/all"), body);
        }

        /// <summary>
        /// Get an invitation code by ID
        /// </summary>
        public async Task<PlusApiResultInvitationCodeVO?> GetByIdAsync(string id)
        {
            return await _client.GetAsync<PlusApiResultInvitationCodeVO>(ApiPaths.BackendPath($"/invitation/code/{id}"));
        }

        /// <summary>
        /// Delete an invitation code
        /// </summary>
        public async Task<PlusApiResultBoolean?> DeleteAsync(string id)
        {
            return await _client.DeleteAsync<PlusApiResultBoolean>(ApiPaths.BackendPath($"/invitation/code/{id}"));
        }
    }
}

File: sdkwork-backend-sdk-swift/Sources/API/InvitationCodeApi.swift
import Foundation

public class InvitationCodeApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Update an existing invitation code
    public func update(body: InvitationCodeForm) async throws -> PlusApiResultInvitationCodeVO? {
        let response = try await client.put(ApiPaths.backendPath("/invitation/code"), body: body)
        return response as? PlusApiResultInvitationCodeVO
    }

    /// Create a new invitation code
    public func create(body: InvitationCodeForm) async throws -> PlusApiResultInvitationCodeVO? {
        let response = try await client.post(ApiPaths.backendPath("/invitation/code"), body: body)
        return response as? PlusApiResultInvitationCodeVO
    }

    /// Get invitation codes by page
    public func listByPage(body: QueryListForm? = nil, params: [String: Any]? = nil) async throws -> PlusApiResultPageInvitationCodeVO? {
        let response = try await client.post(ApiPaths.backendPath("/invitation/code/list"), body: body, params: params)
        return response as? PlusApiResultPageInvitationCodeVO
    }

    /// Get all invitation codes
    public func listAllEntities(body: QueryListForm? = nil) async throws -> PlusApiResultListInvitationCodeVO? {
        let response = try await client.post(ApiPaths.backendPath("/invitation/code/list/all"), body: body)
        return response as? PlusApiResultListInvitationCodeVO
    }

    /// Get an invitation code by ID
    public func getById(id: String) async throws -> PlusApiResultInvitationCodeVO? {
        let response = try await client.get(ApiPaths.backendPath("/invitation/code/\(id)"))
        return response as? PlusApiResultInvitationCodeVO
    }

    /// Delete an invitation code
    public func delete(id: String) async throws -> PlusApiResultBoolean? {
        let response = try await client.delete(ApiPaths.backendPath("/invitation/code/\(id)"))
        return response as? PlusApiResultBoolean
    }
}
